pkg/middleware/requestid/requestid.go

```diff
@@ -12,23 +13,23 @@ import (
 )
 
 const (
-	requestIDCtxKey   = "request-id-context-key"
+	requestIDKey      = "request_id"
 	requestIDTraceKey = "request_id"
 
 	// RequestIDHeader defines the HTTP header that is set in each HTTP response
 	// for a given request. The value of the header is unique per request.
 	RequestIDHeader = "X-Request-Id"
 )
 
-// FromContext extracts the request-id from the context, if it exists.
-func FromContext(ctx context.Context) (string, bool) {
-	if md, ok := metadata.FromOutgoingContext(ctx); ok {
-		if vals := md.Get(requestIDCtxKey); len(vals) > 0 {
-			return vals[0], true
-		}
+// InitID returns the ID to be used to identify the request.
+// If trace is enabled, returns trace ID; otherwise returns a new ULID.
+func InitID(ctx context.Context) string {
```

Comment:
```suggestion
// InitRequestID returns the ID to be used to identify the request.
// If tracing is enabled, returns trace ID; otherwise returns a new UUID.
func InitID(ctx context.Context) string {
```

and let's remove the `// TODO use ulid library?` for now

Reply:
Good catch @miparnisari. Thank your for all your collaboration. 

I've just pushed a change that added the func rename too.